client/components/chip/index.tsx

```diff
@@ -1,14 +1,25 @@
 /** @format **/
+import React from 'react';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import { HoverTooltip } from 'wcpay/components/tooltip';
 
-const types = [ 'primary', 'light', 'warning', 'alert' ];
+const types = [ 'primary', 'light', 'warning', 'alert' ] as const;
 
-const Chip = ( props ) => {
+export type ChipType = typeof types[ number ];
+
+interface Props {
+	message: string;
+	type: ChipType;
+	isCompact?: boolean;
+	className?: string;
+	tooltip?: React.ReactNode;
+}
+
+const Chip: React.FC< Props > = ( props ) => {
 	const { message, type, isCompact, className, tooltip } = props;
```

Comment:
Nit/question: should we prefer to destructure props within the function param list or on a separate line (as is currently implemented)?

```suggestion
const Chip: React.FC< Props > = ( {
	message,
	type,
	isCompact,
	className,
	tooltip
} ) => {
```

Reply:
I usually prefer to do it within the function param list, but don't like too much when it ends up taking so many lines 😄

Reply:
I think it's situational 🤔 I generally favor destructuring the props, but in some cases I think destructuring in the function (or even using `prop.<x>` is better.

At the same time I'm sort of indefferent? 😅  Like here I'd be fine with either. I personally like that the destructuring is on just one line before the change though 👍